python-v2/pypi-assets/formatter.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+# coding=utf-8
+"""Reformat PyPI JSON metadata."""
+import argparse
+import copy
+import json
+
+from functools import reduce
+from urllib.parse import urljoin
+
+
+def main():
+    """Parse arguments, consume stdin, and produce stdout."""
+    parser = argparse.ArgumentParser(description='Prune PyPI JSON metadata.')
+    parser.add_argument(
+        'metadata_path',
+        type=argparse.FileType('r'),
+        help=('''\
+            The path to a PyPI JSON metadata file. If -, the file is read from
+            stdin.
+            '''),
+    )
+
+    subparsers = parser.add_subparsers(
+        title="actions",
+        dest="action",
+        help=('''
+        Actions that can be performed by the formatter: prune_metadata will cut unwanted
+        distributions from the metadta. format_url will replace distribution url links 
+        with a base_url.
+        ''')
+    )
+
+    prune_metadata_parser = subparsers.add_parser("prune_metadata")
+    prune_metadata_parser.add_argument(
+        'distributions',
+        type=json.loads,
+        help='A JSON list of filenames to search for and keep in the metadata.'
+    )
+
+    format_url_parser = subparsers.add_parser("format_url")
+    format_url_parser.add_argument(
+        'base_url',
+        type=str,
+        help='base_url at which the fixture packages are located'
+    )
+
+    args = parser.parse_args()
+
+    if args.action == 'prune_metadata':
+        print(prune_metadata(json.load(args.metadata_path), args.distributions))
+    elif args.action == 'format_url':
+        print(format_url(json.load(args.metadata_path), args.base_url))
```

Comment:
`args.metadata_path` is a file handle, and it's never being explicitly closed, and `json.load(args.metadata_path)` is called twice. We can improve robustness and DRYness with the following code:

```python
    args = parser.parse_args()
    try:
        metadata = json.load(args.metadata_path)
    finally:
        args.metadata_path.close()

    if args.action == 'prune_metadata':
        print(prune_metadata(metadata, args.distributions))
    elif args.action == 'format_url':
        print(format_url(metadata, args.base_url))
```